diff --git a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
index 01c9daa..b04614f 100644
--- a/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
+++ b/node_modules/@dr.pogodin/react-native-fs/ios/ReactNativeFs.mm
@@ -1058,33 +1058,34 @@ - (NSString *)getPathForDirectory:(NSSearchPathDirectory)directory
   return [paths firstObject];
 }
 
-- (facebook::react::ModuleConstants<JS::NativeReactNativeFs::Constants>)constantsToExport
+- (facebook::react::ModuleConstants<JS::NativeReactNativeFs::Constants::Builder>)constantsToExport
 {
-  return facebook::react::typedConstants<JS::NativeReactNativeFs::Constants>({
-    .CachesDirectoryPath = [self getPathForDirectory:NSCachesDirectory],
-    .DocumentDirectoryPath = [self getPathForDirectory:NSDocumentDirectory],
-    .DownloadDirectoryPath = @"",
-    .ExternalCachesDirectoryPath = @"",
-    .ExternalDirectoryPath = @"",
-    .ExternalStorageDirectoryPath = @"",
-    .MainBundlePath = [[NSBundle mainBundle] bundlePath],
-    .TemporaryDirectoryPath = NSTemporaryDirectory(),
-    .FileTypeRegular = NSURLFileResourceTypeRegular,
-    .FileTypeDirectory = NSURLFileResourceTypeDirectory,
-    .DocumentDirectory = 0,
-    .LibraryDirectoryPath = [self getPathForDirectory:NSLibraryDirectory],
-    .PicturesDirectoryPath = @"",
-    .RoamingDirectoryPath = @"",
-    .FileProtectionKeys = JS::NativeReactNativeFs::FileProtectionKeysT::Builder({
-      .FileProtectionComplete = NSFileProtectionComplete,
-      .FileProtectionCompleteUnlessOpen = NSFileProtectionCompleteUnlessOpen,
-      .FileProtectionCompleteUntilFirstUserAuthentication = NSFileProtectionCompleteUntilFirstUserAuthentication,
-      .FileProtectionNone = NSFileProtectionNone,
-    }),
-  });
+  NSDictionary *constants = @{
+    @"CachesDirectoryPath": [self getPathForDirectory:NSCachesDirectory],
+    @"DocumentDirectoryPath": [self getPathForDirectory:NSDocumentDirectory],
+    @"DownloadDirectoryPath": @"",
+    @"ExternalCachesDirectoryPath": @"",
+    @"ExternalDirectoryPath": @"",
+    @"ExternalStorageDirectoryPath": @"",
+    @"MainBundlePath": [[NSBundle mainBundle] bundlePath],
+    @"TemporaryDirectoryPath": NSTemporaryDirectory(),
+    @"FileTypeRegular": NSURLFileResourceTypeRegular,
+    @"FileTypeDirectory": NSURLFileResourceTypeDirectory,
+    @"DocumentDirectory": @0,
+    @"LibraryDirectoryPath": [self getPathForDirectory:NSLibraryDirectory],
+    @"PicturesDirectoryPath": @"",
+    @"RoamingDirectoryPath": @"",
+    @"FileProtectionKeys": @{
+      @"FileProtectionComplete": NSFileProtectionComplete,
+      @"FileProtectionCompleteUnlessOpen": NSFileProtectionCompleteUnlessOpen,
+      @"FileProtectionCompleteUntilFirstUserAuthentication": NSFileProtectionCompleteUntilFirstUserAuthentication,
+      @"FileProtectionNone": NSFileProtectionNone,
+    },
+  };
+  return [_RCTTypedModuleConstants newWithUnsafeDictionary:constants];
 }
 
-- (facebook::react::ModuleConstants<JS::NativeReactNativeFs::Constants>)getConstants
+- (facebook::react::ModuleConstants<JS::NativeReactNativeFs::Constants::Builder>)getConstants
 {
   return [self constantsToExport];
 }
